MPSGameLift/Code/Source/Components/UI/UiGameLiftFlexMatchConnect.cpp

```diff
@@ -0,0 +1,206 @@
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &UiGameLiftFlexMatchConnect::m_attemptConnectionBlockerUi, "Attempt Connection Blocker", "Fullscreen UI for blocking user input while the client tries to connect.")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &UiGameLiftFlexMatchConnect::m_connectToHostFailedUi, "Connection To Host Failed", "UI to inform the user that connecting to the host failed.")
+                    ;
+            }
+        }
+    }
+
+    void UiGameLiftFlexMatchConnect::Activate()
+    {
+        UiCursorBus::Broadcast(&UiCursorInterface::IncrementVisibleCounter);
+
+        // Listen for button presses
+        UiButtonBus::Event(m_quitButtonUi, &UiButtonInterface::SetOnClickCallback, [this](AZ::EntityId buttonEntityId, [[maybe_unused]] AZ::Vector2 position) { OnButtonClicked(buttonEntityId); });
+        UiButtonBus::Event(m_connectButtonUi, &UiButtonInterface::SetOnClickCallback, [this](AZ::EntityId buttonEntityId, [[maybe_unused]] AZ::Vector2 position) { OnButtonClicked(buttonEntityId); });
+        UiButtonBus::Event(m_connectToHostFailedUi, &UiButtonInterface::SetOnClickCallback, [this](AZ::EntityId buttonEntityId, [[maybe_unused]] AZ::Vector2 position) { OnButtonClicked(buttonEntityId); });
+
+        // Hide the attempting connection ui until the player tries to connect
+        UiElementBus::Event(m_attemptConnectionBlockerUi, &UiElementInterface::SetIsEnabled, false);
+
+        // Listen for disconnect events to know if connecting to the host server failed
+        AZ::Interface<Multiplayer::IMultiplayer>::Get()->AddEndpointDisconnectedHandler(m_onConnectToHostFailed);
+        AZ::Interface<IRegionalLatencyFinder>::Get()->AddRequestLatenciesCompleteEventHandler(m_requestLatenciesComplete);
+
+        // Listen for auth events
+        AWSClientAuth::AWSCognitoAuthorizationNotificationBus::Handler::BusConnect();
+
+        // Listen for matchmaking events
+        AZ::Interface<IMatchmaking>::Get()->AddMatchmakingFailedEventHandler(m_onMatchmakingFailed);
+        AZ::Interface<IMatchmaking>::Get()->AddMatchmakingSuccessEventHandler(m_onMatchmakingSuccess);
+        AZ::Interface<IMatchmaking>::Get()->AddMatchmakingTicketReceivedEventHandler(m_onMatchmakingTicketReceived);
+    }
+
+    void UiGameLiftFlexMatchConnect::Deactivate()
+    {
+        AWSClientAuth::AWSCognitoAuthorizationNotificationBus::Handler::BusDisconnect();
+        m_onConnectToHostFailed.Disconnect();
+        UiCursorBus::Broadcast(&UiCursorInterface::DecrementVisibleCounter);
+    }
+
+    void UiGameLiftFlexMatchConnect::OnButtonClicked(AZ::EntityId buttonEntityId)
+    {
+        const auto console = AZ::Interface<AZ::IConsole>::Get();
+        if (!console)
+        {
+            AZ_Assert(false, "UiGameLiftFlexMatchConnect attempting to use console commands before AZ::Console is available.");
+            return;
+        }
+
```

Comment:
This could all move into the quitButton code, the console isn't used by the other 2 buttons.

Reply:
Cleaned up!